python/lib/dependabot/python/update_checker/pip_compile_version_resolver.rb

```diff
@@ -143,9 +144,13 @@ def handle_pip_compile_errors(error)
           end
 
           if error.message.match?(GIT_REFERENCE_NOT_FOUND_REGEX)
-            name = error.message.match(GIT_REFERENCE_NOT_FOUND_REGEX).
-                   named_captures.fetch("name")
-            raise GitDependencyReferenceNotFound, name
+            tag = error.message.match(GIT_REFERENCE_NOT_FOUND_REGEX).named_captures.fetch("tag")
+            constraints_section = error.message.split("Finding the best candidates:").first
+            egg_regex = /#{Regexp.escape(tag)}#egg=(\S+)/
```

Comment:
According to the [Python docs](https://packaging.python.org/en/latest/tutorials/packaging-projects/#configuring-metadata), we have reasonable guarantees for package distribution names (at least for PyPI):

> `name` is the distribution name of your package. This can be any name as long as it only contains letters, numbers, `_` , and `-`. It also must not already be taken on pypi.org.

```suggestion
            egg_regex = /#{Regexp.escape(tag)}#egg=([A-Za-z0-9\_\-]+)/
```

Extracting that into a named constant:

```ruby
# See https://packaging.python.org/en/latest/tutorials/packaging-projects/#configuring-metadata
PYTHON_PACKAGE_NAME_REGEX = /[A-Za-z0-9\_\-]+/
```

```ruby
            egg_regex = /#{Regexp.escape(tag)}#egg=#{PYTHON_PACKAGE_NAME_REGEX}/
```